### PR TITLE
Percentage bars

### PR DESCRIPTION
**Pull request by silasary**

### CODE REVIEW

File: decksite/achievements.py [hunk 1]
```diff
@@ -94,7 +94,6 @@ def display(self, p: 'person.Person') -> str:  # pylint: disable=no-self-use, un
     def load_summary(self, season_id: Optional[int] = None) -> Optional[str]:
         season_condition = f'AND season_id = {season_id}' if season_id != 'all' else ''
         sql = f"""SELECT SUM(`{self.key}`) AS num, COUNT(DISTINCT person_id) AS pnum FROM _achievements WHERE `{self.key}` > 0 {season_condition}"""
-        print(sql)
```
Comment (silasary) [inline decksite/achievements.py:97]: @adambiltcliffe Please don't use print() on decksite.  It's really bad for performance.

File: decksite/main.py [hunk 2]
```diff
@@ -94,6 +94,10 @@ def achievements():
     view = Achievements(auth.mtgo_username())
     return view.page()
 
+@APP.route('/person/achievements/')
```
Comment (silasary) [inline decksite/main.py:97]: @adambiltcliffe Also try not to break old URLs.  
If you change any other URLs in the future, please add a redirect.  (It's bad for SEO, old links, and people's bookmarks)

File: decksite/achievements.py [hunk 3]
```diff
@@ -103,6 +102,14 @@ def load_summary(self, season_id: Optional[int] = None) -> Optional[str]:
             players_text = ngettext('1 player', f'%(num)d players', res.pnum)
             return f'Earned{times_text} by {players_text}.'
         return None
+    def percent(self, season_id: Optional[int] = None) -> float:
+        season_condition = f'season_id = {season_id}' if season_id != 'all' else ''
```
Comment (bakert) [inline decksite/achievements.py:106]: This was the 500 on /seasons/all/achievements – we have `query.season_query(season_id)` to encapsulate this logic.

File: decksite/achievements.py [hunk 4]
```diff
@@ -103,6 +102,14 @@ def load_summary(self, season_id: Optional[int] = None) -> Optional[str]:
...
+        sql = f"""SELECT SUM(CASE WHEN {self.key} > 0 THEN 1 ELSE 0 END) AS pnum, COUNT(*) AS mnum FROM _achievements WHERE {season_condition}"""
```
Comment (bakert) [inline decksite/achievements.py:107]: nit. Triple quotes for multiline strings, single quotes for single-line strings, double quotes for single-line strings containing apostrophes.

File: decksite/templates/achievements.mustache [hunk 5]
```diff
@@ -4,7 +4,12 @@
         {{#achievement_descriptions}}
         <dt class='{{class}}'>{{title}}</dt>
         <dd>{{{description_safe}}}</dd>
-        {{#summary}}<dd>{{summary}}</dd>{{/summary}}
+        {{#summary}}
+            <dd class='achievement_holder'>
```
Comment (bakert) [inline decksite/templates/achievements.mustache:8]: nit. Double quotes for html attribute values.

**Review by bakert — COMMENTED** [review]
This is cool! I think we need to make it a bit clearer what it is but good stake in the ground.

File: decksite/achievements.py (hunk 3 above)
Comment (silasary) [reply decksite/achievements.py:106]: This was me blindly copying the code above after merging with head at midnight

Comment (bakert) [reply decksite/achievements.py:106]: c8e43f7d fixes the above call.

**Pull request merged**